app/src/checkupdatesdialog.cpp

```diff
@@ -0,0 +1,193 @@
+#include "checkupdatesdialog.h"
+#include <QNetworkReply>
+#include <QNetworkAccessManager>
+#include <QHBoxLayout>
+#include <QVBoxLayout>
+#include <QLabel>
+#include <QProgressBar>
+#include <QPushButton>
+#include <QDesktopServices>
+
+#include "util.h"
+
+
+CheckUpdatesDialog::CheckUpdatesDialog()
+{
+    Qt::WindowFlags eFlags = Qt::Dialog | Qt::WindowTitleHint | Qt::WindowCloseButtonHint;
+    setWindowFlags(eFlags);
+    setMinimumSize(QSize(400, 150));
+
+    QLabel* logoLabel = new QLabel;
+    logoLabel->setPixmap(QPixmap(":/icons/logo.png"));
+    logoLabel->setFixedSize(QSize(72, 72));
+
+    mTitleLabel = new QLabel;
+    mTitleLabel->setText(tr("Checking for Updates...", "status description in the check-for-update dialog"));
+
+    mDetailLabel = new QLabel;
+    mDetailLabel->setWordWrap(true);
+
+    //If minimum and maximum both are set to 0, the bar shows a busy indicator instead of a percentage of steps. 
+    mProgressBar = new QProgressBar;
+    mProgressBar->setMaximum(0);
+    mProgressBar->setMinimum(0);
+    mProgressBar->setValue(0);
+
+    mDownloadButton = new QPushButton(tr("Download"));
+    mCloseButton = new QPushButton(tr("Close"));
+
+    QHBoxLayout* hButtonLayout = new QHBoxLayout;
+    hButtonLayout->addWidget(mDownloadButton);
+    hButtonLayout->addWidget(mCloseButton);
+    
+    QVBoxLayout* vLayout = new QVBoxLayout;
+    vLayout->addWidget(mTitleLabel);
+    vLayout->addWidget(mDetailLabel);
+    vLayout->addWidget(mProgressBar);
+    vLayout->addSpacerItem(new QSpacerItem(20, 20, QSizePolicy::Minimum, QSizePolicy::Expanding));
+    vLayout->addLayout(hButtonLayout);
+
+    QHBoxLayout* mainLayout = new QHBoxLayout;
+    mainLayout->addWidget(logoLabel);
+    mainLayout->addLayout(vLayout);
+    setLayout(mainLayout);
+
+    mDownloadButton->setDisabled(true);
+
+    connect(mDownloadButton, &QPushButton::clicked, this, &CheckUpdatesDialog::gotoDownloadPage);
+    connect(mCloseButton, &QPushButton::clicked, this, &CheckUpdatesDialog::closeDialog);
+}
+
+CheckUpdatesDialog::~CheckUpdatesDialog()
+{
+    mNetworkManager->deleteLater();
+}
+
+void CheckUpdatesDialog::startChecking()
+{   
+#ifdef NIGHTLY
+    nightlyBuildCheck();
+#else
+    regularBuildCheck();
+#endif
+}
+
+void CheckUpdatesDialog::regularBuildCheck()
+{
+    mNetworkManager = new QNetworkAccessManager(this);
+    QUrl url("https://www.pencil2d.org/version/");
```

Comment:
It would be better to use https://github.com/pencil2d/pencil/releases.atom. It's XML so it is still easy enough to parse, but it get's updated automatically, which is one less thing for us to forget during an update :wink: